### ISSUE TITLE
Tekton API Reference

### ISSUE THREAD

**Issue by skaegi** [tektoncd/pipeline#1250]
Currently a lot of our doc is examples that do not necessarily reflect all the dark corners of our API. It might be time to craft or generate open API for everything and add a reference API section to our doc.

Something like -- https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.15/

**Comment by duanshiqiang** [comment]
Yes, it would be very useful.

**Comment by bobcatfish** [comment]
Going to close this as a duplicate of https://github.com/tektoncd/pipeline/issues/2055 which has an open PR against it currently and covers the same thing

**Comment by sbwsg** [comment]
I'm going to reopen this one since #2055 ended up in a closed state. If I remember right, though I can't find the specific PR, we decided that a static description of the API wasn't enough and that we wanted to be able to regenerate the API documentation.  This issue describes that requirement a bit more clearly than #2055 so I'm calling this one the source of truth.  Happy to switch it around again if others prefer.

**Comment by sbwsg** [comment]
I've created a new milestone for 0.16 and tentatively put this in there to be discussed for inclusion.

**Comment by dibyom** [comment]
Ref: https://github.com/tektoncd/pipeline/pull/3081 -- starts documenting our API using Open API

**Comment by dibyom** [comment]
We need to publish the generated OpenAPI specs to the website.

**Comment by afrittoli** [comment]
https://github.com/tektoncd/website/issues/194

**Comment by pritidesai** [comment]
hey @afrittoli is there anything outstanding from pipeline perspective with API Spec? this is part of 0.18 and checking what is left with this feature 🙏

**Comment by afrittoli** [comment]
The missing bit is publishing it to the website.
tektoncd/website#194

I've been working on it but I don't think it'll be ready today. I'll try to finish in time.

**Comment by pritidesai** [comment]
thank you @afrittoli 🙏

**Comment by pritidesai** [comment]
this is being tracked in [tektoncd/website#117](https://github.com/tektoncd/website/issues/117)

**Comment by dibyom** [comment]
Remaining work: once tektoncd/website#117 is done, put a link to the API reference in this repo.

**Comment by 06kellyjac** [comment]
Still requires https://github.com/tektoncd/website/issues/117

/remove-lifecycle stale

**Comment by prashant-juluri** [comment]
> Currently a lot of our doc is examples that do not necessarily reflect all the dark corners of our API. It might be time to craft or generate open API for everything and add a reference API section to our doc.
> 
> Something like -- https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.15/

seems inaccessible. any equivalent link?

**Comment by mostafa-eltaher** [comment]
> > Currently a lot of our doc is examples that do not necessarily reflect all the dark corners of our API. It might be time to craft or generate open API for everything and add a reference API section to our doc.
> > Something like -- https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.15/
> 
> seems inaccessible. any equivalent link?

Perhaps https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.23/ which also points to the OpenAPI Specs at https://github.com/kubernetes/kubernetes/blob/release-1.23/api/openapi-spec/swagger.json

**Comment by abayer** [comment]
So #4494 is done - I'm not sure what needs to be done on the website side to have it pick up https://github.com/tektoncd/pipeline/blob/main/docs/pipeline-api.md and link to it?

**Comment by lbernick** [comment]
I think this issue can be closed and the website changes can be tracked in https://github.com/tektoncd/website/issues/363
